### PR TITLE
Add CorrThreeBodyByTwoBodyCondition.

### PR DESCRIPTION
**Pull request by rekovic**
#### PR description:

This is motivated by and replaces  PR #36103. 
There is no overlap-removal logic in this class, but rather a 3-object correlation condition, where correlations are done in pairs of two objects 
- First correlation condition between 
   - leg1 and leg2
- Second correlation condition between 
   - leg3 and leg1
   - leg3 and leg2

 Implementation of the logic motivated by PR #36103.

   Logic:
   - Define 3 GtConditionCategories: (1st & 2nd correlation legs with one correlation condition, and 3rd leg and its correlation condition with 1st and with 2nd leg )

   - Loop leg1: over all objects passing conditions of 1st correlation leg
     - Loop leg2: over all objects passing conditions of 2nd correlation leg
       - Check for dEta, dPhi, dR, and mass (and charge) correlation condition of 1st-leg and 2nd-leg object.
       - Loop over all objects of leg3 passing conditions of 3rd correlation leg
         - Check for dEta, dPhi, dR, correlation condition of 3rd leg with 1st and 2nd leg objects,each.
         - If correlation cuts NOT passing, save objects leg1, leg2, and leg3 object.
       - End loop leg3
     - End loop leg2
   - End loop leg1

#### PR validation:

Ran on Data and MC samples with Testing menu for Run3. 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by rekovic** [comment]
@cmsbuild , please test

**Comment by rekovic** [comment]
+1

**Comment by qliphy** [comment]
+1

File: L1Trigger/L1TGlobal/interface/CorrelationThreeBodyByTwoBodyTemplate.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+#include "L1Trigger/L1TGlobal/interface/GlobalDefinitions.h"
+// forward declarations
+
+// class declaration
+class CorrelationThreeBodyByTwoBodyTemplate : public GlobalCondition {
+public:
+  /// constructor(s)
+  ///   default
+  CorrelationThreeBodyByTwoBodyTemplate();
```
Comment (makortel) [inline L1Trigger/L1TGlobal/interface/CorrelationThreeBodyByTwoBodyTemplate.h:37]: These member functions do not appear to be defined anywhere, and that broke UBSAN build, see https://github.com/cms-sw/cmssw/issues/36561

**Comment by Dr15Jones** [comment]
@cms-sw/orp-l2 this should be reverted as it breaks the UBSAN and generally doesn't make any sense.

**Pull request merged**